### ISSUE TITLE
Limit number of posts

### ISSUE THREAD

**Issue by uGeek** [shouya/rss-funnel#8]
Is there a possibility of reducing the number of posts in the original feed? Adding it in the Filters section?

**Comment by shouya** [comment]
Yes, you can use a JS filter on `feed`:

``` yaml
  - path: /hn-first-5.xml
    source: https://news.ycombinator.com/rss
    filters:
      - js: |
          function modify_feed(feed) {
            feed.items = feed.items.slice(0, 5);
            return feed;
          }
```


May I ask what your real-life use case is? I may implement a standalone filter for this feature if it's suited for general use cases.

**Comment by uGeek** [comment]
It's so that when I use a feed that has about 500 posts, I limit it to 5 or 10.


And is it possible to mix 2 or more RSS feeds?

**Comment by shouya** [comment]
> It's so that when I use a feed that has about 500 posts, I limit it to 5 or 10.

I see. BTW you can also add a [query parameter](https://github.com/shouya/rss-funnel/blob/master/src/server/endpoint.rs#L101) to the endpoint to limit the number of posts. e.g. instead of `http://<addr>/your-endpoint.xml`, query for `http://<addr>/your-endpoint.xml?limit_posts=10`. Please note that this feature is not documented because I'm not certain if I want to change it in the future.

> And is it possible to mix 2 or more RSS feeds?

No, it's not possible at the moment. I've been considering adding this feature since yesterday, and now that you've brought it up as well, I think it's worth implementing. I'll work on adding this feature in the coming days.

**Comment by shouya** [comment]
@uGeek FYI the `merge_feed` filter was implemented in https://github.com/shouya/rss-funnel/pull/18 and will be included in next release (0.0.5).
